openreview/journal/webfield/actionEditorWebfield.js

```diff
@@ -0,0 +1,206 @@
+        var status = {};
+        if (completedReview) {
+          var confidenceString = completedReview.content.confidence.value;
+          var confidence = parseInt(confidenceString.substring(0, confidenceString.indexOf(':')));
+          confidences.push(confidence);
+          status = {
+            Recommendation: completedReview.content.recommendation.value,
+            Confidence: confidence,
+            'Review length': completedReview.content.review.value.length
+          }
+        }
+        reviewerStatus[reviewer.anonId] = {
+          id: reviewer.id,
+          name: view.prettyId(reviewer.id),
+          email: reviewer.id,
+          completedReview: completedReview && true,
+          forum: submission.id,
+          note: completedReview && completedReview.id,
+          status: status
+        }
+      })
+
+      var stats = {
+        Confidence: {
+          min: confidences.length ? _.min(confidences) : 'N/A',
+          max: confidences.length ? _.max(confidences) : 'N/A',
+          avg: confidences.length ? _.round(_.sum(confidences) / confidences.length, 2) : 'N/A'
+        }
+      };
+
+      rows.push({
+        checkbox: { selected: false, noteId: submission.id },
+        submissionNumber: { number: number},
+        submission: { number: number, forum: submission.forum, content: { title: submission.content.title.value, authors: submission.content.authors.value, authorids: submission.content.authorids.value}},
+        reviewProgressData: {
+          numSubmittedReviews: reviews.length,
+          numReviewers: reviewers.length,
+          reviewers: reviewerStatus,
+          stats: stats
+        },
+        actionEditorData: {
+          recommendation: decision && decision.content.recommendation.value,
+          editUrl: decision ? ('/forum?id=' + submission.id + '&noteId=' + decision.id + '&referrer=' + referrerUrl) : null
+        }
+      })
+    }
+  })
+
+
+  return venueStatusData = {
+    invitations: invitations,
+    rows: rows
+  };
+}
+
+// Render functions
+var renderHeader = function() {
+  Webfield.ui.setup('#group-container', VENUE_ID);
+  Webfield.ui.header(HEADER.title, HEADER.instructions);
+
+  var loadingMessage = '<p class="empty-message">Loading...</p>';
+  var tabsList = [
+    {
+      heading: 'Assigned Papers',
+      id: 'assigned-papers',
+      content: loadingMessage,
+      extraClasses: 'horizontal-scroll',
+      active: true
+    },
+    {
+      heading: 'Action Editor Tasks',
+      id: 'action-editor-tasks',
+      content: loadingMessage
+    }
+  ];
+
+  Webfield.ui.tabPanel(tabsList);
+};
+
+var renderTableAndTasks = function(venueStatusData) {
+
+  Webfield2.ui.renderTasks('#action-editor-tasks', venueStatusData.invitations, { referrer: encodeURIComponent('[Action Editor Console](/group?id=' + ACTION_EDITOR_ID + '#action-editor-tasks)')});
+
+  Webfield2.ui.renderTable('#assigned-papers', ['<input type="checkbox" id="select-all-papers">', '#', 'Paper Summary',
```

Comment:
Nice! This looks a lot better than what we are currently using. 

My only suggestion would be to move the headings array and the row render functions array into the options object. So the function signature would be: `renderTable(container, data, options = {})`. I think it will make it even more clear.

Reply:
does it mean the headings and templates are optionals? what should I use if they are not present?

Reply:
I like the idea of having a default template for any object where we can show the key-values in a div.

Reply:
Yeah I was thinking the headings could be optional (not shown if not provided), and the render functions could be like you said, just printing the values.